numcodecs/ndarray_like.py

```diff
@@ -51,6 +51,26 @@ def view(self, dtype: DType = ...) -> "NDArrayLike":
         ...
 
 
+# keep in sync with NDArrayLike
+_NDArrayLike_protocol_attrs = [
```

Comment:
Nice and simple, but do we need all of these? I assume looping through takes some time too. I am surprised not to see the `__array*__` methods, though (NEP18 etc).

Reply:
A subset would probably be enough. I can look through the zarr codebase tomorrow to check what's currently used. I just based it on the NDArrayLike class to be consistent. Currently total accumulated runtime of is_ndarray_like reduces from 14.9 to 2.8 seconds in my benchmark.

Reply:
This has been dropped after adopting Mads' suggestion